internal/bzlmod/go_deps.bzl

```diff
@@ -251,9 +259,37 @@ def _go_deps_impl(module_ctx):
                     ", ".join([str(tag.go_mod) for tag in module.tags.from_file]),
                 ),
             )
+
         additional_module_tags = []
+        from_file_tags = []
+        go_works = []
+        module_tag_path_to_label = {}
+
         for from_file_tag in module.tags.from_file:
```

Comment:
let's leave ourselves and future travelers a nice comment here, describing what is happening and why.